<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Julien Chiquet" />

<meta name="date" content="2020-09-21" />

<title>Homework: PCA for vizu, compression and exploratory analysis \ SNP and MNIST data sets</title>

<script src="site_libs/header-attrs-2.3/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<script src="resources/hideOutput.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="resources/style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MAP573</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="https://github.com/jchiquet/CourseUnsupervisedLearningX">
    <span class="fa fa-github"></span>
     
    
  </a>
</li>
<li>
  <a href="https://moodle.polytechnique.fr/enrol/index.php?id=9404">
    <span class="fa fa-server"></span>
     
    Moodle
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="instructions.html">
    <span class="fas fa fas fa-gear"></span>
     
    Setup
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-chalkboard-teacher"></span>
     
    Lectures/Slides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://github.com/jchiquet/CourseAdvancedR/raw/master/2020_MAP573/R_intro.pdf">Introduction to R</a>
    </li>
    <li class="dropdown-header">Manipulation and Representation</li>
    <li class="divider"></li>
    <li class="dropdown-header">Dimensionality Reduction</li>
    <li class="dropdown-header">Linear methods: PCA and variants</li>
    <li class="dropdown-header">Non linear methods: Kernel PCA, t-SNE, VAE</li>
    <li class="divider"></li>
    <li class="dropdown-header">Clustering</li>
    <li class="dropdown-header">Partitioning: HAC, k-means, Spectral clustering</li>
    <li class="dropdown-header">Model-based clustering: Gaussian mixture model and EM</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-laptop"></span>
     
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tutorials_R.html">R tutorials</a>
    </li>
    <li>
      <a href="tutorial_PCA_budget.html">PCA</a>
    </li>
    <li class="dropdown-header">auto-encoder</li>
    <li class="dropdown-header">Clustering</li>
    <li class="dropdown-header">GGM and EM</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-keyboard"></span>
     
    Homework
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="homework_1_reporting.html">Swirl, Rmarkdown, R basics</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li class="dropdown-header">Data analysis with PCA</li>
    <li class="dropdown-header">Report for lab #2</li>
    <li class="dropdown-header">Spectral Clustering for graphs</li>
    <li class="dropdown-header">Handling Missing Values</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-chart-bar"></span>
     
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Subject 1:</li>
    <li class="dropdown-header">Subject 2:</li>
    <li class="dropdown-header">Subject 3:</li>
    <li class="dropdown-header">Subject 4:</li>
    <li class="dropdown-header">Subject 5:</li>
    <li class="dropdown-header">Subject 6:</li>
    <li class="dropdown-header">Subject 7:</li>
    <li class="dropdown-header">Subject 8:</li>
    <li class="dropdown-header">Subject 9:</li>
    <li class="dropdown-header">Subject 10:</li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Homework: PCA for vizu, compression and exploratory analysis \ SNP and MNIST data sets</h1>
<h4 class="author">Julien Chiquet</h4>
<h4 class="date">2020-09-21</h4>

</div>


<div id="preliminaries" class="section level2">
<h2>Preliminaries</h2>
<div id="information" class="section level3">
<h3>Information</h3>
<p>Data sets are available at <a href="https://github.com/jchiquet/MAP573/tree/master/data"></a>. Homework is due Sunday 10/18 23:59 in Rmd (see assignment in Moodle).</p>
</div>
<div id="package-requirements" class="section level3">
<h3>Package requirements</h3>
<p>We start by loading a couple of packages for data manipulation, dimension reduction and fancy representations.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(tidyverse)    <span class="co"># advanced data manipulation and vizualisation</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="kw">library</span>(knitr)        <span class="co"># R notebook export and formatting </span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="kw">library</span>(FactoMineR)   <span class="co"># Factor analysis</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="kw">library</span>(factoextra)   <span class="co"># Fancy plotting of FactoMineR outputs</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a><span class="kw">library</span>(kableExtra)   <span class="co"># integration of table in Rmarkdown</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co"># set default ggplot2 theme to black and white</span></span></code></pre></div>
</div>
</div>
<div id="snp-data-genotyping-of-various-population" class="section level2 tabset">
<h2 class="tabset">SNP data: genotyping of various population</h2>
<div id="data-description" class="section level3">
<h3>Data description</h3>
<p>A single-nucleotide polymorphism is a substitution of a single nucleotide that occurs at a specific position in the genome, where each variation is present at a level of 0.5% from person to person in the population. They are coded as 0, 1 or 2 (meaning 0, 1 or 2 allels different regarding the reference population)</p>
<p><a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">See the great wikipedia page for detail!</a></p>
<p>We can measure SNP for individuals with high trhoughput technology and SNP array. SNP chips for human contains more than 1 million variables! We only suggest to analyse a sample of a data set containing the 5500 most variant SNP for 728 individuals with various origin, with the following descriptors:</p>
<ul>
<li>CEU: Utah residents with Northern and Western European ancestry from the CEPH collection</li>
<li>GIH: Gujarati Indians in Houston, Texas</li>
<li>LWK: Luhya in Webuye, Kenya</li>
<li>MKK: Maasai in Kinyawa, Kenya</li>
<li>TSI: Toscani in Italia</li>
<li>YRI: Yoruba in Ibadan, Nigeria</li>
</ul>
<p>The data are imported as follows:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="kw">load</span>(<span class="st">&quot;data/SNP.RData&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a>snp  &lt;-<span class="st"> </span>data<span class="op">$</span>Geno <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a><span class="st">  </span><span class="kw">add_column</span>(<span class="dt">origin =</span> data<span class="op">$</span>origin, <span class="dt">.before =</span> <span class="dv">1</span>) </span></code></pre></div>
</div>
<div id="questions" class="section level3">
<h3>Questions</h3>
<ol start="0" style="list-style-type: decimal">
<li>Have a look at the data a make some brief descriptive plot/summary</li>
<li>Fit a PCA on these data. Justify the scaling or not.</li>
<li>Represent the scree plot for the 100 first axes. Comment. Estimate the number of axis with <code>estim_npc</code></li>
<li>Plot individual factor maps on axes 1, 2 and 3. Add colors and ellipses associated with the factor ‘origin’</li>
<li>Check who are the first, say, 250 most contributive individuals to these axes. Show them in the projection. Have a look at the cosine of the most influent guys.</li>
<li>Plot the correlation circle. Only retain variables based on the quality of their representation and/or degree of contribution to the axes represented.</li>
<li>Summarize the above analyses in biplots. Add fancy colors and whatever.</li>
<li>Indicate a group of individual as supplementary (i.e., not use to fit the PC). Show how excluding a group influence (or not) the fit and the projection, by exploring several groups. Explain.</li>
</ol>
</div>
</div>
<div id="mnist-data-handwritten-digit-data" class="section level2 tabset">
<h2 class="tabset">MNIST data: handwritten digit data</h2>
<div id="data-description-1" class="section level3">
<h3>Data description</h3>
<p>The MNIST dataset is an acronym that stands for the Modified National Institute of Standards and Technology dataset. It is a dataset of 60,000 small square 28×28 pixel grayscale images of handwritten single digits between 0 and 9. It is commonly used for training various image processing systems. The database is also widely used for training and testing in the field of machine learning.</p>
<p>The full data set is <a href="mnist_raw%20%3C-%20read_csv(%22https://pjreddie.com/media/files/mnist_train.csv%22,%20col_names%20=%20FALSE)">available here</a>.</p>
<p><strong>But careful!!, it weigths more than 100 Mo so check your internet connection</strong></p>
<p>A sample of 2,000 instances is provided for convenience <a href="&#39;https://jchiquet.github.io/MAP573/data/mnist_sample.csv&#39;">mnist_sample</a>.</p>
</div>
<div id="questions-1" class="section level3">
<h3>Questions</h3>
<ol style="list-style-type: decimal">
<li>Load an format the data (first column is the label) as a <code>tibble</code> or a <code>data.frame</code>. Each row is a vector with size 784 (28 x 28) + a column for the label.</li>
<li>Write a function to plot an 28 x 28 image based on a vector, and test it on a row of your data frame.</li>
<li>Make a PCA, choice wether to scale or not.</li>
<li>Make the usual diagnostic plots (screeplot, individual, variable) for the first axes, and comment. Use GCV to select the number of axes.</li>
<li>Write a function to reconstruct an image with two argument: <span class="math inline">\(i\)</span> (an instance) <span class="math inline">\(k\)</span> (the number of component used).</li>
<li>Plot the reconstructed image for various value of <span class="math inline">\(k\)</span> <!-- 7  Bonus: try kernel PCA with package `kernlab` or t-SNE with package `tsne`. --></li>
</ol>
<!-- ## Decathlon: analysing performance between athletes {.tabset} -->
<!-- _Courtesy of [Julie Josse](http://juliejosse.com/), thanks!_ -->
<!-- ### Data description -->
<!-- This dataset contains the results of decathlon events during two athletic meetings which took place one month apart in 2004: the Olympic Games in Athens (23 and 24 August), and the Decastar 2004 (25 and 26 September). For both competitions, the following information is available for each athlete: performance for each of the 10 events, total number of points (for each event, an athlete earns points based on performance; here the sum of points scored) and final ranking. The events took place in the following order: 100 meters, long jump, shot put, high jump, 400 meters (first day) and 110 meter hurdles, discus, pole vault, javelin, 1500 meters (second day). Nine athletes participated to both competitions. We would like to obtain a typology of the performance profiles. -->
<!-- The data are distributed with {FactoMineR} -->
<!-- ```{r data set} -->
<!-- library(FactoMineR) -->
<!-- data("decathlon") -->
<!-- ``` -->
<!-- ### Questions -->
<!-- 1. Inspect the data set with the following commands: -->
<!-- 2. Explain the interest of centering and scaling the data. Could you spotlight outstanding athletes ? Which inequality are you using? -->
<!-- 3. Explain your choices for the active and illustrative variables/individuals and perform the PCA on this data set. -->
<!-- 4. Comment -->
<!-- - the correlation between the 100 m and long.jump -->
<!-- - the correlation between long.jump and Pole.vault -->
<!-- - can you describe the athlete Casarsa?  -->
<!-- - the proximity between Sebrle and Clay  -->
<!-- - the proximity between Schoenbeck and Barras -->
<!-- 5. In which trials those who win the decathlon perform the best? Could we say that the decathlon trials are well selected?  -->
<!-- 6. Compare and comment the performances during both events: Decastar and Olympic. Could we conclude on the differences?  Plot Confidence ellipses -->
<!-- 7. Select the number of dimensions with the function `estim_npc`.  -->
<!-- ### Solution (compulsive click forbidden!) -->
<!-- 1. Inspect the data set with the following commands: -->
<!-- 2. Explain the interest of centering and scaling the data. Could you spotlight outstanding athletes? -->
<!-- When the data are standardized, it is possible to compare two variables with different units and to say sentences such as "Paul is more remarkable by his height than John is by is weight". When looking at the standardized data, we can look for the values greater or smaller than 2 for instance. We are refering to Bienaymé-Tchebychev which states that 25% of the observations will be at 2 standard deviation from their means. If we consider Gaussian data, we can refine this inequality and consider know that 4.5 % of the observations are greater than 2 in absolute value. Sebrle value for Javeline is 2.528251350 meaning that he is far above average.  -->
<!-- The aim of conducting PCA on this dataset is to determine profiles for similar performances: are there any athletes who are better at endurance events or those requiring short bursts of energy, etc? And are some of the events similar? If an athlete performs well in one event, will he necessarily perform well in another? -->
<!-- 3. Explain your choices for the active and illustrative variables/individuals and perform the PCA on this data set. -->
<!-- ```{r PCA decathlon} -->
<!-- decathlon_PCA <- PCA(decathlon, quanti.sup = c(11,12) , quali.sup = 13) -->
<!-- fviz_eig(decathlon_PCA) -->
<!-- fviz_pca_ind(decathlon_PCA, axes = c(3, 4)) -->
<!-- fviz_pca_var(decathlon_PCA, axes = c(3, 4)) -->
<!-- ``` -->
<!-- To obtain a typology of the athletes based on their performances for the 10 -->
<!-- decathlon events, such as "two athletes are close as they have similar -->
<!-- performance profiles", the distances between two athletes are defined on the -->
<!-- basis of their performances in the 10 events. Thus, only the performance -->
<!-- variables are considered active; the other variables (number of points, rank, -->
<!-- and competition) are supplementary. Here, the athletes are all considered as active -->
<!-- individuals. -->
<!-- 4. Comment:  -->
<!--   - the correlation between the 100 m and long.jump -->
<!-- The 100m and long.jump are negatively correlated: therefore, an athlete who runs 100 meters quickly will generally jump a long way. The variables 100m, 400m, and 110m hurdles are positively correlated, that is, some athletes perform well in all four events while others do not. -->
<!--   - the correlation between long.jump and Pole.vault -->
<!-- Since long.jump is well represented in the first plan and Pole.vault is not, we can deduce that long.jump and Pole.vault are approximately orthogonal, meaning that the corresponding variables are roughly uncorrelated. Overall, the variables relating to speed are negatively correlated with the first principal component while the variables shot put and long jump are positively correlated with this component. The coordinates of these active variables can be found in the object  which also gives the representation quality of the variables (cosine squared) and their contributions to the construction of the components. -->
<!-- Bourguignon and Karpov have very different performance profiles since they are opposed according to the main axis of variability.  -->
<!--   - can you describe the athlete Casarsa?  -->
<!-- Casarsa is located on the top left corner. The first dimension is highly correlated with the number of points: this indicates that he does not have a large number of points. The second dimension is correlated with the Shot.put, High.jump and Discus. This indicates that Casarsa had good results in these three sports. Remember that the second dimension is calculated orthogonally to the first. So Casarsa has good results in these three sports compared to other "bad" athletes.   -->
<!--   - the proximity between Sebrle and Clay  -->
<!-- Sebrle and Clay are close to one another and both far from the center of gravity of the cloud of points. The quality of their projection is therefore good, and we can be certain that they are indeed close in the original space. This means that they have similar profiles in their results across all sports events. -->
<!--   - the proximity between Schoenbeck and Barras -->
<!-- Schoenbeck and Barras are close to one another but they are also close to the center of gravity of the cloud of points. When looking at their cos2 they are not well projected, We cannot interpret their distance based on this plot only. -->
<!-- 5. In which trials those who win the decathlon perform the best? Could we say that the decathlon trials are well selected?  -->
<!-- The supplementary variable "number of points" is almost collinear to the first principal component. Therefore, the athletes with a high number of points are -->
<!-- particularly good in the trials correlated with the first principal component. Those who win the decathlon perform the best in 100m, 110m hurdles and long jump. This means that the ranking of the decathlon is governed by those three sports. -->
<!-- 6. Compare and comment the performances during both events: Decastar and Olympic, by ploting confidence ellipses. -->
<!-- ```{r ellipse} -->
<!-- plotellipses(decathlon_PCA) -->
<!-- ``` -->
<!-- 7. Select the number of dimensions with the function `estim_npc`.  -->
<!-- ```{r} -->
<!-- plot(estim_ncp(decathlon[, 1:10])$criterion, type = "l", xlab = "number of axis", ylab = "GCV") -->
<!-- ``` -->
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
